crates/sui-move-build/src/lib.rs

```diff
@@ -269,6 +272,52 @@ pub fn build_from_resolution_graph(
     })
 }
 
+pub fn resolve_toolchain_version(
+    lock_file: Option<PathBuf>,
+    default_flavor: Option<Flavor>,
+    default_edition: Option<Edition>,
+) -> anyhow::Result<ToolchainVersion> {
+    // Top-level invariant: if a flavor is set (i.e., default_flavor is Some) it must be valid.
+    match default_flavor {
+        Some(flavor) if flavor != Flavor::Sui => anyhow::bail!(
+            "The flavor of the Move compiler cannot be overridden with anything but \
+                        \"{}\", but the default override was set to: \"{flavor}\" in Move.toml",
+            Flavor::Sui
+        ),
+        _ => (),
+    };
+
+    let base_toolchain_version = ToolchainVersion {
+        compiler_version: env!("CARGO_PKG_VERSION").into(),
```

Comment:
Manually tested this resolves correctly, not sure there's a sensible test if this version keeps getting bumped.